baseapp_auth/tests/integration/test_users.py

```diff
@@ -215,6 +224,104 @@ def test_admin_can_only_deactivate_account(self, user_client):
         h.responseNoContent(r)
         assert User.objects.filter(id=user_id, is_active=False).exists() is True
 
+    def test_anonymize_and_delete_is_triggered(self, user_client):
+        user_client.user.is_superuser = False
+        user_client.user.save()
+        user_id = user_client.user.id
+
+        with patch(
+            "baseapp_auth.rest_framework.users.tasks.anonymize_and_delete_user_task.apply_async"
+        ) as mock_apply_async:
+            r = user_client.delete(self.reverse())
+            h.responseNoContent(r)
+            mock_apply_async.assert_called_once_with(
+                args=[user_id], eta=mock_apply_async.call_args[1]["eta"]
+            )
+
+    def test_anonymize_and_delete_user_task_sends_success_email(self, outbox):
+        user = UserFactory()
+        user_id = user.id
+
+        anonymize_and_delete_user_task(user_id)
+        assert len(outbox) == 1
```

Comment:
@mathieubouhelier guess we should cover both scenarios where it should and shouldn't be triggered:

if `config.SEND_USER_ANONYMIZE_EMAIL_TO_SUPERUSERS = True`  and if `config.SEND_USER_ANONYMIZE_EMAIL_TO_SUPERUSERS = False`